File: android/src/main/java/com/lightningkite/khrysalis/net/HttpResponseError.actual.kt
package com.lightningkite.khrysalis.net

open class HttpResponseException(val response: HttpResponse, cause: Throwable? = null): Exception("Got code ${response.code}", cause)
class HttpReadResponseException(response: HttpResponse, val text: String, cause: Throwable? = null): HttpResponseException(response, cause)

File: ios/Khrysalis/src/net/HttpResponseError.actual.swift
import Foundation


class HttpResponseException : Exception {
    let response: HttpResponse
    public init (_ response: HttpResponse, _ cause: Exception? = nil) {
        self.response = response
        super.init("Got response \(response.code)", cause)
    }
    public convenience init (response: HttpResponse, cause: Exception? = nil) {
        self.init(response, cause)
    }
}